### PR TITLE
Add Tempban capabilities to ban

### PR DESCRIPTION
**Pull request by DarkView**
Behaves the same way temporary mutes would, where time is optional after the user argument

### CODE REVIEW

File: backend/src/migrations/1608753440716-CreateTempBansTable.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          },
+          {
+            name: "expires_at",
+            type: "datetime",
+          },
+        ],
+      }),
+    );
+  }
+
```
Comment (Dragory) [inline backend/src/migrations/1608753440716-CreateTempBansTable.ts:35]: Could add an index on `expires_at` here!

File: backend/src/plugins/ModActions/ModActionsPlugin.ts [hunk 2]
```diff
@@ -165,8 +168,13 @@ export const ModActionsPlugin = zeppelinGuildPlugin<ModActionsPluginType>()("mod
 
     state.mutes = GuildMutes.getGuildInstance(guild.id);
     state.cases = GuildCases.getGuildInstance(guild.id);
+    state.tempbans = GuildTempbans.getGuildInstance(guild.id);
     state.serverLogs = new GuildLogs(guild.id);
 
+    state.unloaded = false;
```
Comment (Dragory) [inline backend/src/plugins/ModActions/ModActionsPlugin.ts:174]: Need to also set this to `true` in `onUnload`

File: backend/src/plugins/ModActions/commands/BanCmd.ts [hunk 3]
```diff
@@ -36,16 +45,82 @@ export const BanCmd = modActionsCmd({
     if (!user.id) {
       return sendErrorMessage(pluginData, msg.channel, `User not found`);
     }
+    const time = args["time"] ? args["time"] : null;
+    console.log(time);
```
Comment (Dragory) [inline backend/src/plugins/ModActions/commands/BanCmd.ts:49]: Leftover debugging? 👀

File: backend/src/plugins/ModActions/commands/BanCmd.ts [hunk 4]
```diff
@@ -36,16 +45,82 @@ export const BanCmd = modActionsCmd({
...
+        if (!existingTempban && !time) {
+          sendErrorMessage(pluginData, msg.channel, `User is already banned indefinitely.`);
+          return;
+        }
+
+        // Ask the mod if we should update the existing ban
+        const alreadyBannedMsg = await msg.channel.createMessage("User is already banned, update ban?");
+        const reply = await waitForReaction(pluginData.client, alreadyBannedMsg, ["✅", "❌"], msg.author.id);
+
+        alreadyBannedMsg.delete();
```
Comment (Dragory) [inline backend/src/plugins/ModActions/commands/BanCmd.ts:81]: Could use a `.catch(noop)` (`noop` is from utils I think) so we don't get an error here if someone e.g. deletes the message before reacting.

File: backend/src/plugins/ModActions/functions/banUserId.ts [hunk 5]
```diff
@@ -87,22 +100,31 @@ export async function banUserId(
   // Create a case for this action
   const modId = banOptions.caseArgs?.modId || pluginData.client.user.id;
   const casesPlugin = pluginData.getPlugin(CasesPlugin);
+
+  const noteDetails: string[] = [];
+  const timeUntilUnban = banTime ? humanizeDuration(banTime) : "indefinite";
+  const timeDetails = `Banned ${banTime ? `for ${timeUntilUnban}` : "indefinitely"}`;
+  if (notifyResult.text) noteDetails.push(notifyResult.text);
```
Comment (Dragory) [inline backend/src/plugins/ModActions/functions/banUserId.ts:107]: Could do `ucfirst(notifyResult.text)` here like it was previously

File: dashboard/package.json [hunk 6]
```diff
@@ -6,7 +6,7 @@
   "scripts": {
     "build": "rimraf dist && cross-env NODE_ENV=production webpack --config webpack.config.js",
     "build-debug": "rimraf dist && cross-env NODE_ENV=development webpack --config webpack.config.js",
-    "watch": "cross-env NODE_ENV=development webpack-dev-server",
+    "watch": "cross-env NODE_ENV=development webpack-dev-server --disable-host-check",
```
Comment (Dragory) [inline dashboard/package.json:9]: Probably not needed in this PR

File: backend/src/plugins/ModActions/commands/BanCmd.ts (hunk 3 above)
Comment (DarkView) [reply backend/src/plugins/ModActions/commands/BanCmd.ts:49]: 300px-Perhaps

File: backend/src/plugins/ModActions/commands/BanCmd.ts (hunk 4 above)
Comment (DarkView) [reply backend/src/plugins/ModActions/commands/BanCmd.ts:81]: Also added to the previous ban->forceban upgrade

File: dashboard/package.json (hunk 6 above)
Comment (DarkView) [reply dashboard/package.json:9]: Huh, normally i make sure that isnt included in my PRs 🤔

**Pull request merged**